server/src/test/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitorTests.java

```diff
@@ -50,15 +62,21 @@
 public class WriteLoadConstraintMonitorTests extends ESTestCase {
     public void testRerouteIsCalledWhenAHotSpotIsDetected() {
         final TestState testState = createRandomTestStateThatWillTriggerReroute();
+        final RecordingMeterRegistry recordingMeterRegistry = new RecordingMeterRegistry();
         final WriteLoadConstraintMonitor writeLoadConstraintMonitor = new WriteLoadConstraintMonitor(
             testState.clusterSettings,
             testState.currentTimeSupplier,
             () -> testState.clusterState,
-            testState.mockRerouteService
+            testState.mockRerouteService,
+            recordingMeterRegistry
         );
 
         writeLoadConstraintMonitor.onNewInfo(testState.clusterInfo);
         verify(testState.mockRerouteService).reroute(anyString(), eq(Priority.NORMAL), any());
+
+        // check the test state hotspots are recorded in the counter
+        recordingMeterRegistry.getRecorder().collect();
+        assertMetricsCollected(recordingMeterRegistry, List.of((long) testState.hotspotNodeIds().size()), List.of());
```

Comment:
I kinda think we shouldn't add the assertions around metrics to every test, just to keep those tests focused on the specific aspects that they're testing? I'd be happy to keep it all for the dedicated test that you have at the bottom.